client/src/main/java/org/eclipse/hono/client/device/amqp/AmqpAdapterClientFactory.java

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.client.device.amqp;
+
+import java.util.function.Consumer;
+
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.client.ConnectionLifecycle;
+import org.eclipse.hono.client.HonoConnection;
+import org.eclipse.hono.client.MessageConsumer;
+import org.eclipse.hono.client.impl.AmqpAdapterClientFactoryImpl;
+
+import io.vertx.core.Future;
+
+/**
+ * A Vert.x based factory for creating clients for Hono's AMQP adapter.
+ */
+public interface AmqpAdapterClientFactory extends ConnectionLifecycle<HonoConnection> {
+
+    /**
+     * Creates a new factory for an existing connection.
+     *
+     * @param connection The connection to use.
+     * @param tenantId The ID of the tenant for which the connection is authenticated.
```

Comment:
I am not sure if I understand the purpose of this parameter. My understanding is that the factory will be scoped to the tenant that the auth-id resolves to during authentication of the underlying AMQP connection, or am I mistaken?

Reply:
Unfortunately, the parameter is expected by the parent classes (the constructor in `AbstractSender` requires the tenantId to be not `null`). Of course, this is ugly, because the tenant must match the one used for authentication.
Do you have any ideas on how to get rid of it? Set a dummy or empty string? (would probably only have negative effects for tracing). Read the Auth-Id from the connection and parse it? (wouldn't work for certificate-based authentication). Change the base classes? ...

Reply:
I see